### PR TITLE
fix

### PR DESCRIPTION
**Pull request by baranov-volodymyr**

### CODE REVIEW

**Comment by baranov-volodymyr** [comment]
Screenshot from 2024-08-04 21-24-29

File: pyvisgraph/visible_vertices.py [hunk 1]
```diff
@@ -144,8 +146,11 @@ def edge_in_polygon(p1, p2, graph):
         return False
     if p1.polygon_id == -1 or p2.polygon_id == -1:
         return False
-    mid_point = Point((p1.x + p2.x) / 2, (p1.y + p2.y) / 2)
-    return polygon_crossing(mid_point, graph.polygons[p1.polygon_id])
+    points = points_on_line(p1, p2, EPS)
+    for point in points:
+        if polygon_crossing(point, graph.polygons[p1.polygon_id]):
```
Comment (brswarmer) [inline pyvisgraph/visible_vertices.py:151]: Is using epsilon-approximation actually faster than returning all intersections?

**Pull request merged**